src/Raven.Server/Smuggler/Documents/DatabaseSource.cs

```diff
@@ -108,6 +109,12 @@ public Task<SmugglerInitializeResult> InitializeAsync(DatabaseSmugglerOptionsSer
                 {
                     LastRaftIndex = rawRecord.EtagForBackup;
                 }
+
+                if (options.IgnoreCorruptedDocumentsErrors)
+                {
+                    _database.DocumentsStorage.OnCorruptedDocumentHandler = OnCorruptedDocument;
```

Comment:
this cannot be done like that, because it will be trigger for everything that is using the table at that time, you need to pass it to the GetDocumentsFrom

Reply:
Agree. Then it seems to me that the most suitable location for `OnCorruptedDataHandler` would be directly inside `result`.

Reply:
Then it seems to me that the most suitable location for `OnCorruptedDataHandler` would be directly inside `DatabaseSource` class.